test/use-floating.test.svelte.ts

```diff
@@ -134,5 +135,128 @@ describe('useFloating', () => {
 
 		expect(isPositioned.value).toBe(false);
 	});
-	// TODO: Test all props fallback to default values when reactively set to `undefined`
+	it_in_effect(
+		'fallbacks to default (`bottom`) when `placement` is set to `undefined`',
+		async () => {
+			let placement: Placement | undefined = $state('top');
+
+			const { x, y } = useFloating({
+				...test_config(),
+				middleware: [offset(5)],
+				get placement() {
+					return placement;
+				}
+			});
+
+			// Give time for FloatingUI to calculate the new position
+			await sleep(100);
+
+			expect(x.value).toBe(0);
+			expect(y.value).toBe(-5);
+
+			placement = undefined;
+
+			// Give time for FloatingUI to calculate the new position
+			await sleep(100);
```

Comment:
I've noticed you've used the custom sleep function several times so far. I really dislike the timeout approach you have used. Please consider using the `waitFor` method from testing library
https://testing-library.com/docs/dom-testing-library/api-async/#waitfor

Reply:
Done!
Using `vi.waitFor` instead since it's built into `vitest` itself.